decrypt.go

```diff
@@ -38,6 +38,15 @@ func (p7 *PKCS7) Decrypt(cert *x509.Certificate, pkey crypto.PrivateKey) ([]byte
 			return nil, err
 		}
 		return data.EncryptedContentInfo.decrypt(contentKey)
+
+	case crypto.Decrypter:
+		// Generic case to handle anything that provides the crypto.Decrypter interface.
+		var contentKey []byte
+		contentKey, err := pkey.Decrypt(rand.Reader, recipient.EncryptedKey, nil)
+		if err != nil {
+			return nil, err
+		}
+		return data.EncryptedContentInfo.decrypt(contentKey)
```

Comment:
*rsa.PrivateKey also implements the `crypto.Decrypter` interface, and when you pass nil to the options on an rsa key, Go just call `rsa.DecryptPKCS1v15()`, we can do both of these cases like this:

Your changes look ok, but as `*rsa.PrivateKey` implements decrypter we can merge those two cases with:

```go
	if dec, ok := pkey.(crypto.Decrypter); ok {
		contentKey, err := pkey.Decrypt(rand.Reader, recipient.EncryptedKey, nil)
		if err != nil {
			return nil, err
		}
		return data.EncryptedContentInfo.decrypt(contentKey)
	}

	return nil, ErrUnsupportedAlgorithm
}
```

And there is no need to do `var contentKey []byte` if you are creating it later with `:=`